### PR TITLE
adds tensorboard hparams logging test

### LINKED ISSUE #1217: tensorboard hyperparameters don't update
## 🐛 Bug

Given two sets of HPARAMS, `h_1` and `h_2` where `h_1` is a strict subset of `h_2`. 

1. If you run pytorch lightning with parameters `h_1`, then `h_2`, the additional parameters from `h_2` are not shown in tensorboard
2. If you run pytorch lightning with parameters `h_2`, then `h_1`, the missing parameters from `h_1` are shown empty in tensorboard 

Case 2 is fine, Case 1 is not. 

I already issued this at [tensorboard ](https://github.com/tensorflow/tensorboard/issues/3371) but was directed back here again. 

### To Reproduce
1.  Run code
1. start `tensorboard --logdir=lightning_logs` in same directory
1.  Go to HPARAMS in website
1.  See _**only**_ `layer_1_dim`

#### Code sample

```python
import pytorch_lightning as pl
from argparse import ArgumentParser
import torch

class LitMNIST(pl.LightningModule):
    def __init__(self, hparams):
        super(LitMNIST, self).__init__()
        self.hparams = hparams
        self.layer_1 = torch.nn.Linear(28 * 28, self.hparams.layer_1_dim)

    def forward(self, *args, **kwargs):
        pass



if __name__ == '__main__':
    parser = ArgumentParser()
    parser.add_argument('--layer_1_dim', type=int, default=10)
    args = parser.parse_args()

    # print(args)
    ## > Namespace(layer_1_dim=10)
    model = LitMNIST(hparams=args)
    trainer = pl.Trainer()
    try:
        trainer.fit(model)
    except:
        pass

    parser = ArgumentParser()
    parser.add_argument('--layer_1_dim', type=int, default=10)
    parser.add_argument('--another_hyperparameter', type=int, default=10)
    args = parser.parse_args()

    # print(args)
    ## > Namespace(another_hyperparameter=10, layer_1_dim=10)
    model = LitMNIST(hparams=args)
    trainer = pl.Trainer()
    try:
        trainer.fit(model)
    except:
        pass
```

Change that "solves" the problem: First call net with both parameters
```python
import pytorch_lightning as pl
from argparse import ArgumentParser
import torch

class LitMNIST(pl.LightningModule):
    def __init__(self, hparams):
        super(LitMNIST, self).__init__()
        self.hparams = hparams
        self.layer_1 = torch.nn.Linear(28 * 28, self.hparams.layer_1_dim)

    def forward(self, *args, **kwargs):
        pass



if __name__ == '__main__':
    parser = ArgumentParser()
    parser.add_argument('--layer_1_dim', type=int, default=10)
    parser.add_argument('--another_hyperparameter', type=int, default=10)
    args = parser.parse_args()

    # print(args)
    ## > Namespace(another_hyperparameter=10, layer_1_dim=10)
    model = LitMNIST(hparams=args)
    trainer = pl.Trainer()
    try:
        trainer.fit(model)
    except:
        pass


    parser = ArgumentParser()
    parser.add_argument('--layer_1_dim', type=int, default=10)
    args = parser.parse_args()

    # print(args)
    ## > Namespace(layer_1_dim=10)
    model = LitMNIST(hparams=args)
    trainer = pl.Trainer()
    try:
        trainer.fit(model)
    except:
        pass
```

### Expected behavior
1.  Run code
1.  start `tensorboard --logdir=lightning_logs` in same directory
1.  Go to HPARAMS  in website
1.  See `layer_1_dim` **and** `another_hyperparameter `
    - but `another_hyperparameter` empty in version0

Hackaround:
- Run second code. The trick is to call the net with **all** hyperparameters first, then tensorboard gets  `another_hyperparameter `

### Environment
 - PyTorch Version (e.g., 1.0): `py3.7_cuda101_cudnn7_0`
 - OS (e.g., Linux): `Win10`
 - How you installed PyTorch: `conda `
 - Build command you used (if compiling from source): - 
 - Python version:` 3.7.0`
 - CUDA/cuDNN version: `10.1`
 - GPU models and configuration: `GTX1650`
 - Any other relevant information:
    - `conda list` for pytorch:  `pytorch-lightning         0.7.1                    pypi_0    pypi`

### LINKED ISSUE #2023: Bug in TensorBoardLogger add_hparams()
### Issue
Hyper parameters will not shown in the Tensorboard when using the TensorBoardLogger().

### Bug identification
I am trying to use the add_hparams() function to tune my network. Using the sample code provided in PyTorch, I have no problem to view the hyper parameters and the associated accuracy / loss in the tensorboard. The code is shown below

```
import os

from torch.utils.tensorboard import SummaryWriter

dir_path = "./run2/test_hp"
os.makedirs(dir_path, exist_ok=True)

with SummaryWriter(dir_path) as w:
    for i in range(5):
        w.add_hparams({'lr': 0.1 * i, 'bsize': i},
                      {'hparam/accuracy': 10 * i, 'hparam/loss': 10 * i})
```

But the same code will not work (50% of the time) when I call this code inside the Lightning module. I will explain what the rest of 50% chance is later. The code is shown below. By calling this code alone, I get the following output, guaranteed. 
Selection_079

```
import os
import torch
from torch.nn import functional as F
from torchvision.datasets import MNIST
from torch.utils.data import DataLoader
from torchvision import transforms
from pytorch_lightning import LightningModule
from pytorch_lightning import Trainer
from pytorch_lightning.loggers import TensorBoardLogger
from torch.utils.tensorboard import SummaryWriter


class MyNet(LightningModule):
    def __init__(self):
        super(MyNet, self).__init__()
        self.l1 = torch.nn.Linear(28 * 28, 10)

    def forward(self, x):
        return torch.relu(self.l1(x.view(x.size(0), -1)))

    def configure_optimizers(self):
        return torch.optim.Adam(self.parameters(), lr=0.001)

    def train_dataloader(self):
        dataset = MNIST(os.getcwd(), train=True, download=True, transform=transforms.ToTensor())
        loader = DataLoader(dataset, batch_size=32, num_workers=4, shuffle=True)
        return loader

    def test_dataloader(self):
        dataset = MNIST(os.getcwd(), train=True, download=True, transform=transforms.ToTensor())
        loader = DataLoader(dataset, batch_size=32, num_workers=4, shuffle=False)
        return loader

    def training_step(self, batch, batch_idx):
        x, y = batch
        y_hat = self(x)
        loss = F.cross_entropy(y_hat, y)
        tensorboard_logs = {'train_loss': loss}
        return {'loss': loss, 'log': tensorboard_logs}

    def test_step(self, batch, batch_idx):
        x, y = batch
        y_hat = self(x)
        loss = F.cross_entropy(y_hat, y)
        tensorboard_logs = {'test_loss': loss}
        return {'loss': loss, 'log': tensorboard_logs}

    def test_epoch_end(self, output):
        with SummaryWriter(self.logger.log_dir) as w:
            for i in range(5):
                w.add_hparams({'lr': 0.1 * i, 'bsize': i}, {'hparam/accuracy': 10 * i, 'hparam/loss': 10 * i})
        return {}


dir_path = "."
tb_logger = TensorBoardLogger(dir_path, name='run2')
model = MyNet()
trainer = Trainer(gpus=1, max_epochs=1, logger=tb_logger)
trainer.fit(model)
trainer.test()
```

However, I discover that if I first call my first script, then call the second script. i.e. create the "test_hp" folder first, then create the "version_0" folder. I can see the hyperparameters in the tensorboard. Very weird. And now, if you delete the "test_hp" folder, the hyperparameters disappear again (back to the previous image). **So it seems like the hyperparameters logged inside the Lightning model, is dependent on the data generated by the original PyTorch code.** Otherwise, it cannot display properly, even we are using the same code. Is this a bug? 
Selection_081

### Environment

* CUDA:
	- GPU:
		- Best GeForce In The World
	- available:         True
	- version:           10.2
* Packages:
	- numpy:             1.18.1
	- pyTorch_debug:     False
	- pyTorch_version:   1.5.0
	- pytorch-lightning: 0.7.6
	- tensorboard:       2.2.1
	- tqdm:              4.46.0
* System:
	- OS:                Linux
	- architecture:
		- 64bit
		- ELF
	- processor:         x86_64
	- python:            3.8.2
	- version:           #102-Ubuntu SMP Mon May 11 10:07:26 UTC 2020

### PR DESCRIPTION
**Pull request by williamFalcon**
Fixes #1217 
Fixes #2023

### CODE REVIEW

File: tests/loggers/test_tensorboard.py [hunk 1]
```diff
@@ -1,9 +1,48 @@
...
+import tests.base.utils as tutils
+from pytorch_lightning import Trainer
+from pytorch_lightning.callbacks import ModelCheckpoint
 from pytorch_lightning.loggers import TensorBoardLogger
+from tests.base import EvalModelTemplate
+from tensorboard.backend.event_processing.event_accumulator import EventAccumulator
+import yaml
+
+
+@pytest.importorskip('torch', minversion='1.5.0')
```
Comment (williamFalcon) [inline tests/loggers/test_tensorboard.py:16]: @Borda is this correct? can you modify so we can merge

Comment (Borda) [inline tests/loggers/test_tensorboard.py:16]:
```suggestion
@pytest.mark.importorskip('torch', minversion='1.5.0')
```

Comment (Borda) [reply tests/loggers/test_tensorboard.py]: it seems it is not among marks so we need to do standard `skipif`
https://docs.pytest.org/en/stable/reference.html#marks
